docs/gke/iap.md

```diff
@@ -0,0 +1,272 @@
+the [enable_iap_openapi.sh](https://github.com/google/kubeflow/tree/master/docs/gke/enable_iap.sh) script.
+
+
+```
+CLIENT_ID=<Client id for OAuth client created in the previous step>
+CLIENT_SECRET=<Client secret for OAuth client created in the previous step>
+SERVICE=envoy
+INGRESS=envoy-ingress
+enable_iap_openapi.sh ${PROJECT} ${NAMESPACE} ${SERVICE} ${INGRESS}
+```
+The above command will output the audience such as:
+
+```
+JWT_AUDIENCE=/projects/991277910492/global/backendServices/801046342490434803
+```
+ 
+ * you will need JWT_AUDIENCE in the next step to configure JWT validation
+
+**Important** Redeploying iap-ingress (e.g. running `ks apply ${ENVIRONMET} -c iap-ingress` again) 
+will cause the JWT_AUDIENCE to change and the backend service created by GCP to change.
+As a result you will have to repeat the steps below to properly configure IAP.
+
+### Deploy Envoy Proxies
+
+The next step is to deploy Envoy as a reverse proxy.
+
+```
+ks generate iap-envoy iap-envoy --audiences=${JWT_AUDIENCE}
+ks apply ${ENVIRONMENT} -c iap-envoy
+```
+
+### Test ingress
+
+It can take some time for IAP to be enabled. You can test things out using the following URL
+
+```
+https://${DOMAIN}/noiap/whoami
+```
+
+  * This a simple test app that's deployed as part of the iap-envoy component.
+  * This URL will always be accessible regardless of whether IAP is enabled or not
+  * If IAP isn't enabled the app will report that you are an anyomous user
+  * Once IAP takes effect you will have to authenticate using your Google account and the app will tell you 
+    what your email is.
+  
+We also have the app running at
+
+```
+https://${DOMAIN}/whoami
+```
+  * But this path will reject traffic that didn't go through IAP so it won't work unless IAP is enabled. In this case you should get
+  one of the following errors.
+
+    * `401 UNAUTHORIZED` this indicates IAP isn't enabled so the request is rejected because it wasn't authenticated
+    * `401 ISS_AUD_UNMATCH1` this is because Envoy doesn't support IAP JWT tokens yet [istio/proxy/issues/941](https://github.com/istio/proxy/issues/941)
+
+
+### Disable JWT verification 
+
+Until [istio/proxy/issues/941](https://github.com/istio/proxy/issues/941) is fixed you can disable JWT verification in Envoy. 
+
+**Warning** This is a serious security risk because it means if IAP is disabled all traffic will be allowed through. You should only do this
+if you understand the risks; for more info see [IAP's docs on Securing Your App](https://cloud.google.com/iap/docs/signed-headers-howto).
+
+```
+ks param set iap-envoy disableJwtChecking true
+ks apply ${ENVIRONMENT} -c iap-envoy
+# Delete the pods so they are recreated with the new config
+kubectl delete pods --selector=service=envoy
+```
+
+### Configure Jupyter to use your Google Identity
+
+We can configure Jupyter to use the identity provided by IAP. This way users won't have to login to JupyterHub.
+
+```
+ks param set ${CORE_NAME} jupyterHubAuthenticator iap
```

Comment:
Couldn't see where this CORE_NAME var is set.

Reply:
Added instructions listing environment variables and describing how to set them.